component/org.wso2.carbon.identity.oauth2.grant.token.exchange/src/main/java/org/wso2/carbon/identity/oauth2/grant/token/exchange/utils/TokenExchangeUtils.java

```diff
@@ -749,4 +856,71 @@ private static boolean validateUsingCertificate(SignedJWT signedJWT, IdentityPro
         // At this point 'verifier' will never be null;
         return signedJWT.verify(verifier);
     }
+
+    public static User getLocalUser(OAuthTokenReqMessageContext tokReqMsgCtx,
+                                    String subjectIdentifier) throws IdentityOAuth2Exception {
+
+        User localUser = null;
+        AbstractUserStoreManager userStoreManager = getUserStoreManager(tokReqMsgCtx);
+        try {
+             if(userStoreManager.isExistingUser(subjectIdentifier)) {
+                 localUser = userStoreManager.getUser(null, subjectIdentifier);
+             }
+        } catch (UserStoreException e) {
+            handleException("Error while resolving local user for subject: " + subjectIdentifier);
+        }
+        return localUser;
+    }
+
+    private static String resolveSubjectIdentifier(JWTClaimsSet claimsSet, String[] lookupAttributes)
+            throws IdentityOAuth2Exception {
+
+        String subjectIdentifier = null;
+        for (String lookupAttribute : lookupAttributes) {
+            Object subjectIdentifierObj = claimsSet.getClaim(lookupAttribute);
+            if (subjectIdentifierObj instanceof String) {
+                subjectIdentifier = (String) subjectIdentifierObj;
```

Comment:
we can simply return from here, which will simplify the code later in the method

Reply:
addressed with [4f8ccec](https://github.com/wso2-extensions/identity-oauth2-grant-token-exchange/pull/19/commits/4f8ccecff35a7fd0c1a8486fb8f56a12e1f2b8c7)